bin/ruby-build

```diff
@@ -423,8 +423,15 @@ http_head_aria2c() {
 }
 
 http_get_aria2c() {
-  # shellcheck disable=SC2086
-  log_command aria2c --allow-overwrite=true --no-conf=true --console-log-level=warn --stderr $ARIA2_OPTS -o "$2" "$1" 2>&3
+  local destfile="$2"
+  if [[ $destfile == /* ]]; then
+    # the "-o" option to aria2c cannot be an absolute path, but we can achieve that with "--dir"
+    # shellcheck disable=SC2086
+    log_command aria2c --allow-overwrite=true --no-conf=true --console-log-level=warn --stderr $ARIA2_OPTS --dir "${destfile%/*}" -o "${destfile##*/}" "$1" 2>&3
+  else
+    # shellcheck disable=SC2086
+    log_command aria2c --allow-overwrite=true --no-conf=true --console-log-level=warn --stderr $ARIA2_OPTS -o "$destfile" "$1" 2>&3
+  fi
```

Comment:
Makes sense to me, and confirmed it works with "file", "subdir/file" and "/abs/path".
You could reduce a bit of duplication:
```suggestion
  local destfile="$2"
  local args=()

  if [[ $destfile == /* ]]; then
    # the "-o" option to aria2c cannot be an absolute path, but we can achieve that with "--dir"
    args+=(--dir "${destfile%/*}" -o "${destfile##*/}")
  else
    args+=(-o "$destfile")
  fi

  # shellcheck disable=SC2086
  log_command aria2c --allow-overwrite=true --no-conf=true --console-log-level=warn --stderr $ARIA2_OPTS "${args[@]}" "$1" 2>&3
```

Reply:
@rwstauner Thanks for the deduplication suggestion, but I find that I prefer seeing the full `aria2c` invocation line in both instances even if there is a little duplication, mostly for the ease of reading.